homework#3

Danik2021 opened the pull request:
do header and main section

Danik2021 committed:
homework#3

do header and main section

Danik2021 committed:
homework#3

Доробив все позиціювання

xrustics commented on css/style.css:
```diff
@@ -10,6 +10,8 @@
   font-family: var(--main-font-family);
   color: var(--title-text-color);
+  width: 1600px;
+  outline: 3px solid #b1b1bb1b;
```
Розумію, можливо тобі треба було це для розробки (outline), сам так роблю іноді) але код в ПР - це код готового, виконаного завдання, та в ньому немає бути зайвих елементів. Якщо це не використовується - прибери) Якщо це так треба - то напиши будь ласка для чого саме?)

Danik2021 replied:
Це було для комфорту, Видаляю

xrustics commented on css/style.css:
```diff
@@ -79,93 +96,117 @@ a.phone {
+  padding-right: 215px;
+  padding-left: 215px;
+  padding-top: 60px;
+  padding-bottom: 60px;
```
Якщо задаєш падінги для чотирьох сторін - то краще через властивість padding: 60px 215px. Це набагато менше місця в коді та більш зручно читається

Danik2021 replied:
✔✔✔

xrustics commented on css/style.css:
```diff
@@ -36,6 +38,19 @@ a.phone {
+  padding-right: 215px;
+}
+
+.nav_menu .nav_li + .nav_li {
```
Поясни будь ласка чому саме так використав селектор?)
Питаю бо просто це не звична конструкція ".nav_li + .nav_li"

Danik2021 replied:
Я так зробив перше, тому що в мене тоді буде у всіх марджин зліва окрім першого елементу. А друге, я такий крутий як Репета))

xrustics commented on css/style.css:
```diff
@@ -44,8 +59,10 @@ a.phone {
 
-.logo {
+.logo,
+.footer-logo {
```
Дивись, виходить у нас є 2 класи, що по факту означають те саме - логотип. Розкажи чому саме потрібно 2 класи?) Я просто хочу зрозуміти твою логіку та можливо підкажу як можна зробити краще

Danik2021 replied:
Це було помилка, не логіка) я вже виправив

xrustics commented on css/style.css:
```diff
@@ -79,93 +96,117 @@ a.phone {
+
   font-family: var(--main-font-family);
   font-weight: 500;
   font-size: 16px;
```
Не тільки кольори можна в константи) я б цей розмір шрифту теж виніс би в змінну, бо бачу він використовується декілька разів в документі. Коли думаєш чи треба його в змінну - дивись чи використовується це значення ще десь, і якщо так, то в змінну його)

xrustics replied:
Зазвичай в змінних тримають основні значення кольорів, розміри шрифтів, розміри основних відступів, тощо. Під словом "основні" я маю на увазі коли ці значення використовується по декілька разів в коді css

Danik2021 replied:
✔✔✔

xrustics commented on index.html:
```diff
@@ -74,7 +78,7 @@ <h3>Сучасні Технології</h3>
 
       <section class="section-block">
-        <h2>Чим ми займаємося</h2>
+        <h2 class="paragraph">Чим ми займаємося</h2>
```
Смислово ця назва класу трохи протиречить використанню) Виходить ми використовуємо тег що відповідає за "заголовки" абйому надаємо основний клас з назвою "параграфу". Це можна легко сплутати з тегом <p>  по сенсу. Краще заміни назву на більш підходящою, наприклад "heading" ( перекладається як заголовок), "або з якоюсь приставкою, типу "team- heading" тощо

Danik2021 replied:
✔✔✔

Danik2021 committed:
bugFix Ruslan

 - add in css var(--main-font-size)

Danik2021 committed:
bugFix Ruslan

- add in css var(--main-font-size)
- delete class footer-logo
- h2 have new name heading

Danik2021 committed:
bugFix error

!

xrustics commented:
«A1» У корені проекту є папка **images** з зображеннями.
Переіменуй папку з image на images та всі імпорти зображеннь в проекті відповідно

xrustics commented:
<img width="1175" alt="Знімок екрана 2024-03-05 о 12 57 56" src="https://github.com/Danik2021/goit-markup-hw-01/assets/59570719/ecebb1c2-4832-4ce8-bf17-17fddd607faa">

xrustics commented:
<img width="1440" alt="Знімок екрана 2024-03-05 о 13 00 57" src="https://github.com/Danik2021/goit-markup-hw-01/assets/59570719/08928414-2a13-4cbd-8bc8-f791eac018c0">

Danik2021 committed:
bugFix Ruslan

- add div.container to all section and main blocks
- header fix margin
- fix button in section one

xrustics commented on css/style.css:
```diff
@@ -5,13 +5,20 @@
   --accent-color: #2196f3;
```
«A3» стилі написані в файлі` styles.css`, який знаходиться в папці css
переіменуй style -> styles

Danik2021 replied:
✔✔✔

xrustics commented on images/frontend_developer.svg:
назва файлу потрібно щоб теж відповідала стилю як і по всьому проєкту. Якщо вирішив писати через дефіс - то краще всюди вже через дефіс і писати. Або "-" або "_"

Danik2021 replied:
✔✔✔

xrustics commented on images/product_designer.svg:
назва файлу потрібно щоб теж відповідала стилю як і по всьому проєкту. Якщо вирішив писати через дефіс - то краще всюди вже через дефіс і писати. Або "-" або "_"

Danik2021 replied:
✔✔✔

xrustics commented on images/ui_designer.svg:
назва файлу потрібно щоб теж відповідала стилю як і по всьому проєкту. Якщо вирішив писати через дефіс - то краще всюди вже через дефіс і писати. Або "-" або "_"

Danik2021 replied:
✔✔✔

xrustics commented on index.html:
Переглянь всі теги <link...
там багато дублікатів

Danik2021 replied:
✔✔✔

xrustics commented on css/style.css:
```diff
@@ -5,13 +5,20 @@
 body {
   font-family: var(--main-font-family);
   color: var(--title-text-color);
+  width: 1600px;
```
на цьому все центрування пішло "по сраці" 😄
Виходить ти обмежив боди до 1600px і вже в цих 1600 центруєш контент. А що якщо у користувача більший екран ніж 1600 ? Тоді виходить у користувача буде боди прижатий о лівого краю та мати ширину 1600, а в цих 1600 вже буде центрування. Подумай сам як це можна виправити. Ось скрін:
<img width="1393" alt="Знімок екрана 2024-03-09 о 11 16 58" src="https://github.com/Danik2021/goit-markup-hw-01/assets/59570719/fb43c39f-a373-4fa3-b7d6-d5f0e1069e9e">

Danik2021 replied:
✔✔✔

xrustics commented on css/style.css:
```diff
@@ -5,13 +5,20 @@
   font-family: var(--main-font-family);
   color: var(--title-text-color);
+  width: 1600px;
+  height: 2346px;
```
так не можна робити! прибери height
Ти не можеш обмежувати весь боди по висоті, це анті патерн, краще ніколи так не роби

xrustics replied:
«B12» В елементів відсутня фіксована висота, вона визначається їх контентом.

Danik2021 replied:
✔✔✔

xrustics commented on index.html:
```diff
@@ -29,104 +29,128 @@
     </header>
 
     <main>
       <section class="main-section">
```
Не бачу виконання цього пункту:
«B15» Для всіх секцій використовується один клас .section, задані верхні і нижні падінги по 94px, що відсувають контент всередину секції.
якщо треба додати якісь інші стилі секції - це можна зробити додатковим класом

Danik2021 replied:
Щось не можу зрозуміти що це потрібно виправити

xrustics replied:
Тут схоже як з класом контейнер тільки робиш класс . section  в ньому стилі padding верхній і нижній по 94 px , потім просто додаєш класс на секцію в html. Ну і якщо ці падіннги десь в іншому селекторі задавав, то там треба підчистити

xrustics commented:
Зображень немає на сторінці портфоліо...
Я знаю в чому річ, але не скажу тобі)
Подумай чого так, потім напиши мені чому так в коментарі, та виправ цю штуку)

![image](https://github.com/Danik2021/goit-markup-hw-01/assets/59570719/d270ddaa-2d4e-40d5-8fd8-2e32b46f8461)

xrustics commented:
Завжди перевіряй свою роботу перед сдачею на перевірку)

xrustics commented:
Перевір виконання цього завдання:
«B17» У картках на сторінці Портфоліо є рамка (властивість border), але тільки у нижній частині картки.

Danik2021 committed:
bugFix Ruslan

- images work
- delete in body fixed width
- document css have right name
- folder with pictures have right name

xrustics commented on css/styles.css:
```diff
@@ -57,16 +74,23 @@ a.phone {
 }
 
+.footer > .container > .logo > a {
+  color: #ffffff;
```
В тебе ж наче вже є змінна для цього коліру, використай її

Danik2021 replied:
✔✔✔

Danik2021 committed:
bugFix Ruslan

!

Danik2021 committed:
bugFix Ruslan

- add class .section

xrustics commented on index.html:
```diff
@@ -26,107 +20,131 @@
-            <p>Значимість цих проблем настільки очевидна, що реалізація планових завдань.</p>
-          </li>
-        </ul>
+      <section class="section-block section">
```
Бачу що в тебе всі секції з цими двома класами. Можливо варто об'єднати тоді 2 класи в один section?

Danik2021 replied:
✔✔✔

xrustics commented on portfolio.html:
```diff
@@ -29,86 +23,110 @@
-        </li>
-      </ul>
+    <main class="main">
+      <div class="container">
```
Це не просто div, тут має бути семантично правильний тег. Подумай який та заміни

Danik2021 replied:
✔✔✔

xrustics commented on portfolio.html:
```diff
@@ -29,86 +23,110 @@
+    <main class="main">
+      <div class="container">
+        <ul class="filter">
+          <li class="list"><button type="button" class="radio button">Усі</button></li>
```
Це трохи неправильно давати класс list для тегів li. Чому так?
Тег ul розшифровується як "unordered list"  тобто "не нумерований список". Тут ключове слова "list" що означає"список ". А тег li розшифровується як "list item" що в перекладі "елемент чи одиниця списку". Виходить коли ти задаєш тегу li класс list то таке враження на перший погляд що там має бути в кожному li ще по списку)

xrustics replied:
Якщо треба назвати класс для елементу списку називай так щоб було зрозуміло що воно таке) Ось найкращі імена класу для li --->
item, list-item

Danik2021 replied:
✔✔✔

xrustics commented on index.html:
```diff
@@ -26,107 +20,131 @@
+        </nav>
+        <div class="header-menu">
+          <a class="email link" href="mailto:info@devstudio.com">info@devstudio.com</a>
+          <a class="phone link" href="tel:+380961111111">+38 096 111 11 11</a>
```
А чому б тобі одразу в css не зробити замість класу link одразу просто на теги "a" кинути стиль text-decoration: none;
Це облегчить тобі потім життя

Danik2021 replied:
✔✔✔

Danik2021 committed:
bugFix Ruslan
